On 32-bit, usize as f64 recommends f64::from(usize)

<!--
Hi there! Whether you've come to make a suggestion for a new lint, an improvement to an existing lint or to report a bug or a false positive in Clippy, you've come to the right place.

For bug reports and false positives, please include the output of `cargo clippy -V` in the report.

Thank you for using Clippy!

Write your comment below this line: -->
On i686, the cast_lossless lint recommends using `f64::from(len)` instead of `len as f64` where `len` is of type `usize`. This would <del>change the behaviour on x86_64</del> fail to compile.

We ran into this as well when compiling to `wasm32-unknown-unknown`:

```sh
warning: casting `usize` to `f64` may become silently lossy if you later change the type
  --> world-benchmark/src/lib.rs:94:28
   |
94 |                         / (self.ns_history.len() as f64 * f64::from(n))
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `f64::from(self.ns_history.len())`
   |
   = note: requested on the command line with `-W clippy::cast-lossless`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cast_lossless
```

following this suggestion leads to:

```
error[E0277]: the trait bound `f64: std::convert::From<usize>` is not satisfied
  --> world-benchmark/src/lib.rs:94:28
   |
94 |                         / (f64::from(self.ns_history.len()) * f64::from(n))
   |                            ^^^^^^^^^ the trait `std::convert::From<usize>` is not implemented for `f64`
   |
   = help: the following implementations were found:
             <f64 as std::convert::From<f32>>
             <f64 as std::convert::From<i16>>
             <f64 as std::convert::From<i32>>
             <f64 as std::convert::From<i8>>
           and 135 others

For more information about this error, try `rustc --explain E0277`.
```

would be great fox fix this in the lint, we do find it useful and works well for all other code we have.

--- a/clippy_lints/src/casts/cast_lossless.rs
+++ b/clippy_lints/src/casts/cast_lossless.rs
@@ -93,7 +93,7 @@ fn should_lint(
             } else {
                 64
             };
-            from_nbits < to_nbits
+            !is_isize_or_usize(cast_from) && from_nbits < to_nbits
         },
         (false, true) if matches!(cast_from.kind(), ty::Bool) && meets_msrv(msrv.as_ref(), &msrvs::FROM_BOOL) => true,
         (_, _) => {

--- a/tests/ui/cast_size_32bit.stderr
+++ b/tests/ui/cast_size_32bit.stderr
@@ -14,25 +14,11 @@ LL |     x0 as f64;
    |
    = note: `-D clippy::cast-precision-loss` implied by `-D warnings`
 
-error: casting `isize` to `f64` may become silently lossy if you later change the type
-  --> $DIR/cast_size_32bit.rs:15:5
-   |
-LL |     x0 as f64;
-   |     ^^^^^^^^^ help: try: `f64::from(x0)`
-   |
-   = note: `-D clippy::cast-lossless` implied by `-D warnings`
-
 error: casting `usize` to `f64` causes a loss of precision on targets with 64-bit wide pointers (`usize` is 64 bits wide, but `f64`'s mantissa is only 52 bits wide)
   --> $DIR/cast_size_32bit.rs:16:5
    |
 LL |     x1 as f64;
    |     ^^^^^^^^^
-
-error: casting `usize` to `f64` may become silently lossy if you later change the type
-  --> $DIR/cast_size_32bit.rs:16:5
-   |
-LL |     x1 as f64;
-   |     ^^^^^^^^^ help: try: `f64::from(x1)`
 
 error: casting `isize` to `f32` causes a loss of precision (`isize` is 32 or 64 bits wide, but `f32`'s mantissa is only 23 bits wide)
   --> $DIR/cast_size_32bit.rs:17:5

